From dc90525fc1a0200a6b719d435db74c5aa88a2ef3 Mon Sep 17 00:00:00 2001
From: Daniel P. Berrange <berrange@redhat.com>
Date: Wed, 8 Jul 2009 14:47:14 +0100
Subject: [PATCH 03/16] Add a QXL graphics card type to domain XML schema

* src/qemu_conf.c: Add dummy entry in enumeration
* docs/schemas/domain.rng: Add 'qxl' as a type for the <video> tag
* src/domain_conf.c, src/domain_conf.h: Add QXL to video type
  enumerations
---
 docs/schemas/domain.rng |    1 +
 src/conf/domain_conf.c  |    3 ++-
 src/conf/domain_conf.h  |    1 +
 src/qemu/qemu_conf.c    |    3 ++-
 4 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/docs/schemas/domain.rng b/docs/schemas/domain.rng
index 5e5c4bd..314908b 100644
--- a/docs/schemas/domain.rng
+++ b/docs/schemas/domain.rng
@@ -1049,6 +1049,7 @@
               <value>vmvga</value>
               <value>xen</value>
               <value>vbox</value>
+              <value>qxl</value>
             </choice>
           </attribute>
           <optional>
diff --git a/src/conf/domain_conf.c b/src/conf/domain_conf.c
index 3cd43eb..41ac6d8 100644
--- a/src/conf/domain_conf.c
+++ b/src/conf/domain_conf.c
@@ -198,7 +198,8 @@ VIR_ENUM_IMPL(virDomainVideo, VIR_DOMAIN_VIDEO_TYPE_LAST,
               "cirrus",
               "vmvga",
               "xen",
-              "vbox")
+              "vbox",
+              "qxl")
 
 VIR_ENUM_IMPL(virDomainInput, VIR_DOMAIN_INPUT_TYPE_LAST,
               "mouse",
diff --git a/src/conf/domain_conf.h b/src/conf/domain_conf.h
index 5c64a47..68da297 100644
--- a/src/conf/domain_conf.h
+++ b/src/conf/domain_conf.h
@@ -441,6 +441,7 @@ enum virDomainVideoType {
     VIR_DOMAIN_VIDEO_TYPE_VMVGA,
     VIR_DOMAIN_VIDEO_TYPE_XEN,
     VIR_DOMAIN_VIDEO_TYPE_VBOX,
+    VIR_DOMAIN_VIDEO_TYPE_QXL,
 
     VIR_DOMAIN_VIDEO_TYPE_LAST
 };
diff --git a/src/qemu/qemu_conf.c b/src/qemu/qemu_conf.c
index 8a1d19f..4b49bad 100644
--- a/src/qemu/qemu_conf.c
+++ b/src/qemu/qemu_conf.c
@@ -92,7 +92,8 @@ VIR_ENUM_IMPL(qemuVideo, VIR_DOMAIN_VIDEO_TYPE_LAST,
               "cirrus",
               "vmware",
               "", /* no arg needed for xen */
-              "" /* don't support vbox */);
+              "", /* don't support vbox */
+              "", /* Not implemented QXL yet */);
 
 int qemudLoadDriverConfig(struct qemud_driver *driver,
                           const char *filename) {
-- 
1.6.6.1

